module/zfs/vdev_mirror.c

```diff
@@ -588,6 +589,16 @@ vdev_mirror_io_start(zio_t *zio)
 			 */
 			for (c = 0; c < mm->mm_children; c++) {
 				mc = &mm->mm_child[c];
+
+				/*
+				 * Possible the VDEV was attached in near past
+				 * and resilvering is not finished yet for it,
+				 * so we need to skip to avoid CHKSUM errors
+				 */
+				if (vdev_dtl_contains(mc->mc_vd, DTL_MISSING,
+				    zio->io_txg, 1))
+					continue;
```

Comment:
This makes sense, but I'd like to understand why the `!mm->mm_resilvering` conditional above isn't sufficient to prevent this.  The logic in `vdev_mirror_map_init()` should be setting `mm->mm_resilvering` to prevent exactly this, but it appears there's a case which isn't covered.

Reply:
"mm_resilvering" means a resilvering in progress, but the bug occurs if a scrub in progress

Reply:
If the only the scrub, and not resilver, case which is the problem I think the above comment is pretty misleading and needs to be updated to make it clear why we need this check.

Reply:
I think ```resilvering is not finished yet for it``` needs to be replaced by ```resilvering is not started yet for it```. Will it be more clear?

Reply:
I'd also like to request this small clarification in the comment, as it had me scratching my head similarly to Brian.